### PR TITLE
Add tab complete for flag arguments

### PR DESCRIPTION
**Pull request by mrasnake-doppler**
Add tab completion and suggestions to various flags throughout the CLI.

closes ENG-2592

### CODE REVIEW

File: pkg/cmd/run.go [hunk 1]
```diff
@@ -674,7 +674,9 @@ func init() {
 	forwardSignals := !isatty.IsTerminal(os.Stdout.Fd())
 
 	runCmd.Flags().StringP("project", "p", "", "project (e.g. backend)")
+	runCmd.RegisterFlagCompletionFunc("project", projectIDsValidArgs)
 	runCmd.Flags().StringP("config", "c", "", "config (e.g. dev)")
+	runCmd.RegisterFlagCompletionFunc("config", configNamesValidArgs)
```
Comment (Piccirello) [inline pkg/cmd/run.go:679]: I think autocomplete would be really nice for the `name-transformer` and `mount-template` flags on this command.

File: pkg/cmd/configs_tokens.go [hunk 2]
```diff
@@ -189,25 +189,35 @@ func configTokenSlugsValidArgs(cmd *cobra.Command, args []string, toComplete str
...
+	configsTokensGetCmd.RegisterFlagCompletionFunc("project", projectIDsValidArgs)
 	configsTokensGetCmd.Flags().StringP("config", "c", "", "config (e.g. dev)")
+	configsTokensGetCmd.RegisterFlagCompletionFunc("config", configNamesValidArgs)
 	configsTokensCmd.AddCommand(configsTokensGetCmd)
 
 	configsTokensCreateCmd.Flags().String("name", "", "service token name")
 	configsTokensCreateCmd.Flags().StringP("project", "p", "", "project (e.g. backend)")
+	configsTokensCreateCmd.RegisterFlagCompletionFunc("project", projectIDsValidArgs)
 	configsTokensCreateCmd.Flags().StringP("config", "c", "", "config (e.g. dev)")
+	configsTokensCreateCmd.RegisterFlagCompletionFunc("config", configNamesValidArgs)
```
Comment (Piccirello) [inline pkg/cmd/configs_tokens.go:209]: Would be really nice to autocomplete `access`

File: pkg/cmd/secrets.go [hunk 3]
```diff
@@ -584,37 +584,49 @@ func secretNamesValidArgs(cmd *cobra.Command, args []string, toComplete string)
...
 	secretsDeleteCmd.Flags().StringP("config", "c", "", "config (e.g. dev)")
+	secretsDeleteCmd.RegisterFlagCompletionFunc("config", configNamesValidArgs)
 	secretsDeleteCmd.Flags().Bool("raw", false, "print the raw secret value without processing variables")
 	secretsDeleteCmd.Flags().BoolP("yes", "y", false, "proceed without confirmation")
 	secretsCmd.AddCommand(secretsDeleteCmd)
 
 	secretsDownloadCmd.Flags().StringP("project", "p", "", "project (e.g. backend)")
+	secretsDownloadCmd.RegisterFlagCompletionFunc("project", projectIDsValidArgs)
 	secretsDownloadCmd.Flags().StringP("config", "c", "", "config (e.g. dev)")
+	secretsDownloadCmd.RegisterFlagCompletionFunc("config", configNamesValidArgs)
```
Comment (Piccirello) [inline pkg/cmd/secrets.go:629]: Same comment but for `name-transformer` here

File: pkg/cmd/completion.go [hunk 4]
```diff
@@ -73,7 +73,7 @@ var completionInstallCmd = &cobra.Command{
 		var buf bytes.Buffer
 		var path string
 		var name string
-
+
```
Comment (Piccirello) [inline pkg/cmd/completion.go]: Erroneous addition of whitespace. You'll want to configure your IDE to strip extra whitespace on save.

**Review by Piccirello — COMMENTED** [review]
Really nice job with this. I'm glad you were able to reuse so much existing code. I identified a few additional flags that would benefit from autocomplete.

In terms of commits:
- You'll want to prepend "chore:" to the "add IDE section to gitignore" commit so that it doesn't show up in the changelog
- Prepand "Add" to "completion for Project Flags" commit for consistency

File: pkg/cmd/run.go (hunk 1 above)
Comment (mrasnake-doppler) [reply pkg/cmd/run.go:679]: I figured out the `name-transformer` just fine, but I don't have complete confidence in the business logic for `mount-template`, is that the results you had in mind?

Comment (Piccirello) [reply pkg/cmd/run.go:679]: Yes! That looks great.

File: pkg/cmd/completion.go [hunk 5]
```diff
@@ -73,7 +73,6 @@ var completionInstallCmd = &cobra.Command{
 		var buf bytes.Buffer
 		var path string
 		var name string
-
```
Comment (Piccirello) [inline pkg/cmd/completion.go]: Erroneous whitespace change

**Review by Piccirello — COMMENTED** [review]
One last cleanup and then this is ready to go!

File: pkg/cmd/completion.go (hunk 5 above)
Comment (mrasnake-doppler) [reply pkg/cmd/completion.go]: this is the whitespace you asked me to remove up here: https://github.com/DopplerHQ/cli/pull/379#discussion_r1158041154

Comment (Piccirello) [reply pkg/cmd/completion.go]: Based on the diff, I wouldn't expect this file to change in this PR. My previous comment was noting that whitespace was added. The whitespace is now removed, but so is the entire newline. Once fixed, I would expect this file to be omitted from the diff.

**Comment by Piccirello** [comment]
Can you squash all but the first commit into one commit named "Add completion support for flag arguments"? There should then be two total commits.

**Review by Piccirello — APPROVED** [review]

**Pull request merged**